FlowCrypt/Functionality/Services/EnterpriseServerApi.swift

```diff
@@ -32,6 +32,9 @@ extension EnterpriseServerApiError: LocalizedError {
 class EnterpriseServerApi: EnterpriseServerApiType {
 
     private enum Constants {
+        /// 404 - Not Found, -1001 - request timed out, 503 - service unavailable
+        static let getToleratedStatuses = [404, -1001, 503]
```

Comment:
Please strictly follow this decision chart:

```
call https://fes.<domain>/api/, process the response:
-> did I receive a response and it's status 200, it's a JSON and has the right service value? -> use FES to grab OrgRules
-> did I receive a response and it's status 404? -> don't use FES -> empty OrgRules
-> did I receive a response and it's other status 4xx or 5xx? -> show error to user
-> did I NOT receive a response (server down or timeout) and the internet is otherwise working? -> don't use FES, empty OrgRules
-> did I NOT receive a response (server down or timeout) and the internet is not working? -> show error to user
```

503 falls under `did I receive a response and it's other status 4xx or 5xx? -> show error to user`

Reply:
so how may I understand if `server down`?

Reply:
In Swift, I'm not sure. But generally, it should fail right away, typically with unknown host:

```
$ curl https://unknownnonexistingserver.com
curl: (6) Could not resolve host: unknownnonexistingserver.com
```

Reply:
I'm not sure if one can differentiate `server down` from `timeout`. Most servers that are not there, if I try to curl them, I'm getting a timeout. I think that it would be possible to get a response that says "nothing is running here" but I'm not sure what that would look like.

Anyway, 503 counts as 4xx and 5xx, the only tolerated http status, other than 200, is 404.

To summarize:
 - 503 counts as 5xx
 - for now we can assume that `server down` is the same as timeout
 - need to ensure that "could not resolve host" is also handled